docs/source/scenarios/vm_scheduling.rst

```diff
@@ -0,0 +1,315 @@
+    The ``DecisionPayload`` of the new requirement event only differs in the remaining buffer time from the 
+    old ones.
+  * If the time is exhausted, the simulator will note it as a failed allocation.
+
+See the detailed attributes `here <#action>`_.
+
+Runtime Simulation
+^^^^^^^^^^^^^^^^^^^
+
+Dynamic Utilization
+~~~~~~~~~~~~~~~~~~~~
+
+To make the simulated environment closest to the real situation. We also simulate CPU utilization of each
+VM. The CPU utilization of the VM varies every tick based on the real VM workload readings. 
+We will regularly calculate the total resources (CPU utilization) of each PM in every tick and update 
+to the PM workload for the following decision.
+
+Real-time Energy Consumption
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One of the most important characteristics that cloud providers concern is the PM's enery consumption. As we
+mention before, the lower energy consumption of the PMs, the lower cost to maintain the physical servers. In 
+our simulation, we currently simulate the energy based on the CPU utilization. In short, PM cost more energy
+if it has higher CPU utilization. [`Reference <https://dl.acm.org/doi/10.1145/1273440.1250665>`_]
+
+VM Deallocation
+^^^^^^^^^^^^^^^^
+
+The MARO simulator regularly checks the finished VMs in every tick. It will then release the finished VM's
+resources, including CPU cores and memory, and finally remove the VM from the PM.
+
+Topologies
+-----------
+
+To provide samples from easy to difficult, two kinds of simple topologies are designed and 
+provided in VM Scheduling scenario. 
+
+Azure Topologies
+^^^^^^^^^^^^^^^^^
+
+The original data are provided by `Azure public dataset 
+<https://github.com/Azure/AzurePublicDataset>`_. In our scenario, we pre-processed the AzurePublicDatasetV2. 
+The dataset contains real Azure VM workloads, including the information of VMs and their utilization readings 
+in 2019 lasting for 30 days. The original dataset contains 2,695,548 VMs.
+
+The detailed information of the data schema can be found
+`here <https://github.com/Azure/AzurePublicDataset/blob/master/AzurePublicDatasetV2.md>`_. After pre-processed,
+we only retain real VM creation and deletion time (converted to the tick, 1 tick means 5 minutes in real time),
+VM cores and memory(GB) requirements, and we also renumber the original VM ID.
+As for the utilization readings part, we store the renumbered VM ID and VM's CPU utilization sorting by the timestamp (tick).
+
+**azure.2019.10k**\ : We randomly sampled 10,000 VMs from the AzurePublicDatasetV2.
+
+**azure.2019.336k**\ : We randomly sampled 336,000 VMs from the AzurePublicDatasetV2.
+
+Naive Baseline
+^^^^^^^^^^^^^^^
+
+Belows are the final environment metrics of the method random allocation and best-fit allocation in 
+different topologies. For each experiment, we setup the environment and test for a duration of 30 days.
```

Comment:
the PM setting is listed as a single column, you can mention it and why we do this here.

To test under different resource supply/PM capacity?